fix(buffers): Prevent negative buffer size and event gauges (#23453)
vectordotdev/vector: changelog.d/23453_prevent_negative_buffer_size_and_event_gauges.fix.md

> fixes a bug where the vector_buffer_byte_size and vector_buffer_events gauges could report negative values under certain race conditions.

Reviewer: ```suggestion
Fixed race condition bug causing negative values to be reported by the `vector_buffer_byte_size` and `vector_buffer_events` gauges.
```
